### PR TITLE
Sign by Keystone

### PR DESCRIPTION
**Pull request by renfengshi**
- Update conneting Keystone
- Sign amino
- Sign direct
- Sign ethereum

### CODE REVIEW

File: packages/background/package.json [hunk 1]
```diff
@@ -28,6 +28,8 @@
     "@ethersproject/hash": "^5.7.0",
     "@ethersproject/transactions": "^5.7.0",
     "@ethersproject/wallet": "^5.7.0",
+    "@ethereumjs/tx": "^3.5.2",
+    "@ethereumjs/common": "^2.6.5",
```
Comment (HeesungB) [inline packages/background/package.json:32]: You should use `ethers.js`. Let me know if there is issue

File: packages/background/package.json [hunk 2]
```diff
@@ -41,6 +43,12 @@
     "@ledgerhq/hw-transport": "^6.20.0",
     "@ledgerhq/hw-transport-webhid": "^6.20.0",
     "@ledgerhq/hw-transport-webusb": "^6.20.0",
+    "@keystonehq/bc-ur-registry-cosmos": "^0.0.2-alpha.1",
+    "@keystonehq/bc-ur-registry-eth": "^0.11.4",
+    "@keystonehq/base-eth-keyring": "^0.6.4",
+    "@keystonehq/cosmos-keyring": "^0.0.1-alpha.0",
+    "@keystonehq/bc-ur-registry": "^0.5.1",
+    "@cosmjs/crypto": "^0.29.3",
```
Comment (HeesungB) [inline packages/background/package.json:51]: In Keplr, we do not use @cosmjs in package.json.

File: package.json [hunk 3]
```diff
@@ -96,6 +96,7 @@
     "@types/react": "^16.14.4",
     "@iov/crypto": "2.1.0",
     "axios": "^0.27.2",
+    "@keystonehq/bc-ur-registry": "^0.5.1",
```
Comment (HeesungB) [inline package.json:99]: Is there any reason for using this library in global?

File: packages/background/package.json (hunk 1 above)
Comment (renfengshi) [reply packages/background/package.json:32]: `
import {
  AccessListEIP2930Transaction,
  FeeMarketEIP1559Transaction,
  Transaction,
  TypedTransaction,
} from "@ethereumjs/tx";
`
`ethers.js` does not provide these.

File: packages/hooks/tsconfig.json [hunk 4]
```diff
@@ -5,7 +5,8 @@
     "outDir": "build",
     "declaration": true,
     "rootDir": "src",
-    "jsx": "react"
+    "jsx": "react",
+    "skipLibCheck": true
```
Comment (HeesungB) [inline packages/hooks/tsconfig.json:9]: You should remove this option.

File: tsconfig.json [hunk 5]
```diff
@@ -59,6 +59,7 @@
     "experimentalDecorators": true,        /* Enables experimental support for ES7 decorators. */
     // "emitDecoratorMetadata": true,         /* Enables experimental support for emitting type metadata for decorators. */
 
-    "resolveJsonModule": true
+    "resolveJsonModule": true,
+    "skipLibCheck": true
```
Comment (HeesungB) [inline tsconfig.json:63]: You should remove this option.

**Comment by HeesungB** [comment]
When signing `ether mint`, we use eip-712 standard.
 
 so You should add this code in `stores/src/account/cosmos.ts`

from
 ```
 const eip712Signing = useEthereumSign && this.base.isNanoLedger;
 ```
 
 to
 
 ```
const eip712Signing =
      useEthereumSign && (this.base.isNanoLedger || this.base.isKeystone);
 ```

File: packages/background/package.json (hunk 2 above)
Comment (renfengshi) [reply packages/background/package.json:51]: How can i use `@cosmjs/crypto`?

File: package.json (hunk 3 above)
Comment (renfengshi) [reply package.json:99]: This can delete.

File: packages/hooks/tsconfig.json (hunk 4 above)
Comment (renfengshi) [reply packages/hooks/tsconfig.json:9]: This can be removed.

File: tsconfig.json (hunk 5 above)
Comment (renfengshi) [reply tsconfig.json:63]: If remove this, some packages will cause errors.

**Comment by Thunnini** [comment]
@renfengshi There was too much work and time passed. For various reasons, I made other commits from branch and cherry-picked the later work. Perhaps for this reason the PR was not automatically closed. Because there are so many pieces of work, it was not possible to check them all. If there is something missing, please fix it based on the current develop branch and make a PR again.
Thank you for your contribution.

**Comment by joshteng** [comment]
Why closed both #553 and this

**Pull request closed without merging**